src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -980,12 +980,21 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 
 		try {
 
-			// Desktop DND (Import file)
+			// External File DND (Import file)
 			if (data instanceof NativeDragAndDropData) {
-				if (isWeb) {
-					const browserUpload = this.instantiationService.createInstance(BrowserFileUpload);
-					await browserUpload.upload(target, originalEvent);
-				} else {
+				// Native DND from Desktop
+				if (originalEvent.dataTransfer?.types.indexOf('Files') !== -1) {
```

Comment:
I don't really understand the logic here. `BrowserFileUpload` was supposed to be for web and `NativeFileImport` for desktop based applications. Now we seem to use desktop based import even in web in certain cases? Why can we not implement the support in `BrowserFileUpload`?